Added 0. Get import status of deployment option

mikeblakeuk opened the pull request:
Added method to get the import status to help resolve
"Microsoft.Rest.HttpOperationException: Operation returned an invalid status code 'NotFound'"

mikeblakeuk committed:
Update Program.cs

Added method to get the import status to help resolve
"Microsoft.Rest.HttpOperationException: Operation returned an invalid status code 'NotFound'"

druttka commented on ProvisionSample/Program.cs:
```diff
@@ -276,6 +277,39 @@ static void Main(string[] args)
+                        if (importResult == null)
+                        {
+                            Console.ForegroundColor = ConsoleColor.Yellow;
+                            Console.WriteLine("Import state of {0} is not found!", datasetId);
```
Is the intent to write the attempted importId here instead of the datasetId?

druttka commented on ProvisionSample/Program.cs:
```diff
@@ -295,6 +329,40 @@ static void Main(string[] args)
             }
         }
 
+        static async Task<Import> GetStatus(string workspaceCollectionName, string workspaceId, string importId)
```
Not a big deal, but could we name this GetImport instead, as it's returning the full Import object? That would also be less ambiguous if we added new methods later that got the status of other types.

druttka commented on ProvisionSample/Program.cs:
```diff
@@ -295,6 +329,40 @@ static void Main(string[] args)
+                    return null;
+                }
+
+                Console.WriteLine("Name:     {0}", import.Name);
```
I don't have a strong opinion here, but my preference would be that the Get\* method just returns the Import and defers writing output to the caller.

The new case 0 for getting the status of an import is writing the ImportState, since that is its concern. A user who selects case 0 might be surprised to see so much more output. 

We could add a case 'a' to do a debug dump of the full import object. Alternatively, if this was meant for debugging purposes, we could either remove this output or use Debug.WriteLine instead.

mikeblakeuk committed:
Updates to GetImport from PR#25

Fixed variable name of dataset (changed to importid)
Changed Method to GetImport
Moved Console out logic to main case statement
Remove exception handling from method
Write full exception out

(Note, debug is no good as Operations use the tool without Visual Studio)

druttka commented:
Thanks @mikeblakeuk. This looks good to me. @mattmazzola?

mattmazzola commented on ProvisionSample/Program.cs:
```diff
@@ -79,6 +79,7 @@ static void Main(string[] args)
                 Console.WriteLine("7. Update connection string info for an existing dataset");
                 Console.WriteLine("8. Retrieve a list of Datasets published to a workspace");
                 Console.WriteLine("9. Delete a published dataset from a workspace");
+                Console.WriteLine("0. Get import status of deployment");
```
I think the phrasing of this new option could be confusing/misleading to others.  I don't think deployment makes sense in the context of Importing/Uploading a .pbix file. 

Can you please change the text to: `0. Get status of import` ?

mikeblakeuk committed:
Changed menu option to "0. Get status of import"

Based on PR#25 feedback from @mattmazzola

mikeblakeuk committed:
Removed exit form zero option

druttka commented on ProvisionSample/Program.cs:
```diff
@@ -319,7 +319,6 @@ static void Main(string[] args)
                                 Console.WriteLine("\t{0}: {1}", report.Name, report.WebUrl);
                             }
                         }
-                        break;
```
I believe this `break;` was appropriate and that we should keep it. It appears that the Run block was written such that an exit flag is initialized to false, and it's checked after the try/catch to call Run again, allowing the user to perform another operation. 

I don't believe this will compile with control falling through from one case (0) to another (default). If it did, we'd still end up flipping the exit flag to true, resulting in an immediate exit instead of a repeated prompt to perform another operation.

Am I missing anything here?

mikeblakeuk committed:
Added break back in
